lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1737,6 +1738,17 @@ class SemanticsObject {
     }());
     return result;
   }
+
+  bool _isDisposed = false;
+
+  void dispose() {
+    assert(!_isDisposed);
```

Comment:
This assert goes away at build time, should this also `if (_isDisposed) return;`? Or is this only expected to fail at dev time?

Reply:
That's a good question. I'm just following an established convention. This pattern is used everywhere else. I'd be on board to revisit it, but that's probably outside the scope of this PR?

Reply:
Definitely out of scope, I didn't realize this pattern was the standard :P